comp/core/workloadmeta/component.go

```diff
@@ -129,6 +129,10 @@ type Component interface {
 	// - EventTypeUnset: one for each entity that exists in the store but is not
 	// present in newEntities.
 	Reset(newEntities []Entity, source Source)
+
+	// Push allows external sources to push events to the metadata store.
+	// Only EventTypeSet and EventTypeUnset event types are allowed.
+	Push(events ...PushEvent) error
```

Comment:
(Same reply to all comments regarding type alias)

My intention with this alias is the following:

To an external component, workloadmeta is a store to which it can subscribe to receive events. 
Internally, the store is populated by collectors. An external component doesn't have a clue about collectors, all it receives is an `Event` via the eventBundle channel. 

Given this, as `CollectorEvent` is in direct relation with `Collectors`, I don't see `CollectorEvent` used any where in the codebase other than in workloadmeta internal collectors (which makes sense given that `Collectors` should be transparent/hidden from external components). 

My intent with the type alias is to keep this abstraction preserved. If I had kept this as `CollectorEvent`, it would have worked functionally, but I don't believe it is necessary for external components to know any thing about existence of what we call `Collector`.

The worthiness of added value of this alias is arguable, and I don't have a strong opinion about this.

WDYT?

Reply:
I used the `workloadmeta.Event` type along with a common `Source` passed as a separate argument. This way we preserve hiding the existence of `Collectors` without needing to define a new type/alias.